daemon/api_model_test.go

```diff
@@ -249,28 +250,30 @@ func (s *apiSuite) TestGetModelHasSerialAssertion(c *check.C) {
 	d.overlord.AddManager(deviceMgr)
 	st := d.overlord.State()
 	st.Lock()
+	defer st.Unlock()
 	assertstatetest.AddMany(st, s.storeSigning.StoreAccountKey(""))
 	assertstatetest.AddMany(st, s.brands.AccountsAndKeys("my-brand")...)
 	s.mockModel(c, st, theModel)
 
-	// in case the name of the serial ever changes, just get it state in State
-	// currently it's hard-coded to always be serialserial
-	var authStateData auth.AuthState
-	err = st.Get("auth", &authStateData)
-	c.Assert(err, check.IsNil)
 	serial, err := s.brands.Signing("my-brand").Sign(asserts.SerialType, map[string]interface{}{
 		"authority-id":        "my-brand",
 		"brand-id":            "my-brand",
 		"model":               "my-old-model",
-		"serial":              authStateData.Device.Serial,
+		"serial":              "serialserial",
 		"device-key":          string(encDevKey),
 		"device-key-sha3-384": deviceKey.PublicKey().ID(),
 		"timestamp":           time.Now().Format(time.RFC3339),
 	}, nil, "")
 	c.Assert(err, check.IsNil)
 	assertstatetest.AddMany(st, serial)
+	devicestatetest.SetDevice(st, &auth.DeviceState{
+		Brand:  "my-brand",
+		Model:  "my-old-model",
+		Serial: "serialserial",
+	})
 
 	st.Unlock()
+	defer st.Lock()
```

Comment:
a comment around this would be helpful, as it is it reads rather confusingly if you don't remember that a few lines above we defer'd an unlock and also that go defers are LIFO

Reply:
I understand the concern, but we do this kind of thing in many places,  I'm not sure we want a comment in all of them

Reply:
okay fair enough, it's just a bit difficult to track the state of a stack of `defer` statements and I haven't personally seen anywhere else where we have these multiple `defer` statements, but I'll take your word for it